app/models/group.rb

```diff
@@ -331,6 +332,17 @@ def self.refresh_automatic_groups!(*args)
     args.each { |group| refresh_automatic_group!(group) }
   end
 
+  def self.refresh_has_messages!
+    exec_sql <<-SQL
+      UPDATE groups g SET has_messages = false
+      WHERE NOT EXISTS (SELECT tag.id
```

Comment:
I would also add a `WHERE has_messages = true` so we don't update the rows for no reason ;)

Reply:
Good idea! Made it so.